src/Serilog/Events/LogEventProperty.cs

```diff
@@ -67,7 +68,7 @@ internal LogEventProperty(EventProperty property)
         /// </summary>
         /// <param name="name">The name to check.</param>
         /// <returns>True if the name is valid; otherwise, false.</returns>
-        public static bool IsValidName(string name) => !string.IsNullOrWhiteSpace(name);
+        public static bool IsValidName([NotNullWhen(true)] string? name) => !string.IsNullOrWhiteSpace(name);
```

Comment:
Not sure the attribute makes sense here?

Reply:
without this we get build errors, since the compiler cant derive the nullability

```
  Logger.cs(1369, 65): [CS8604] Possible null reference argument for parameter 'name' in 'LogEventProperty MessageTemplateProcessor.CreateProperty(string name, object? value, bool destructureObjects = false)'.
```

Reply:
Wow! TIL, thanks :+1: